parser.cpp

```diff
@@ -30,21 +30,80 @@ Result<std::unique_ptr<ast::Str>> Parser::parseStr() {
 }
 
 Result<std::unique_ptr<ast::Out>> Parser::parseOut() {
+  Result<std::unique_ptr<ast::Str>> sres = parseStr();
+  if (sres.hasError()) return sres;
+  std::unique_ptr<ast::Str> &str = sres.get();
+  return Result<std::unique_ptr<ast::Out>>(
+      std::make_unique<ast::Out>(std::move(str)));
+}
+
+Result<std::unique_ptr<ast::Expr>> Parser::parseExpr() {
   Result<Token> res = lexer_.Lex();
   if (res.hasError()) return res;
+  Token &tok = res.get();
```

Comment:
For consistency and to avoid potential use-after-frees in the future, let's have this be a value instead of reference: `Token tok = res.get();`